src/components/Sidebar.tsx

```diff
@@ -0,0 +1,30 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { usePostsContext } from '../hooks/usePostsContext';
+import { PostDetails } from './Posts/PostDetails';
+
+export const Sidebar: FC = () => {
+  const { selectedPost } = usePostsContext();
```

Comment:
Ensure that the usePostsContext hook is providing the selectedPost state and it's being updated properly in the context, otherwise your sidebar will not render properly.